src/react/utils.js

```diff
@@ -95,6 +95,31 @@ export function valueIsClassComponent(realm: Realm, value: Value): boolean {
   return false;
 }
 
+export function valueIsReactLibraryObject(realm: Realm, value: ObjectValue): boolean {
+  if (realm.react.reactLibraryObject === value) {
+    return true;
+  }
+  // we check that the object is the React or React-like library by checking for
+  // core properties that should exist on it
+  let reactVersion = Get(realm, value, "version");
```

Comment:
Calling `Get` may execute arbitrary code (a getter), and we haven't established yet that this object is well-behaved. To avoid state mutations or exceptions at this point, wrap the `Get` into a `tryQuery` call (accessible from the ResidualHeapVisitor via `this.logger.tryQuery`). That will prevent such issues.

And note that there's quite an overhead... It's probably beneficial to short-circuit all this somehow.